Python/sysmodule.c

```diff
@@ -129,7 +153,15 @@ _PySys_GetOptionalAttr(PyObject *name, PyObject **value)
         *value = NULL;
         return 0;
     }
-    return PyDict_GetItemRef(sysdict, name, value);
+    int ret = PyDict_GetItemRef(sysdict, name, value);
+#ifndef ABIFLAGS
+    if (ret == 0 && _PyUnicode_EqualToASCIIString(name, "abiflags")) {
+        if (WarnIncomingSysAbiflagsChange() < 0) {
+            return -1;
```

Comment:
Shouldn't we clear value here? because callers to _PySys_GetOptionalAttr() may not have the same expectations then (namely, we're returning -1 but *value* is set) [we should still DECREF it]